### DISCUSSION TITLE
Customizing the /users/me/ query after login

### DISCUSSION THREAD

**Discussion by kadlinobit** [Intevel/nuxt-directus#37] [Q&A] [answered]
Hi, is there a way to customize the call to the `/users/me` endpoint that happens automatically after login and fetches current user data? 

I need to get some related table fields which are not returned by default, thus I would like to add some attributes to the call, e.g. `/users/me?fields=*,profiles.*`.

**Comment by Intevel** [comment, marked as answer]
Hey @kadlinobit.

Currently there is no way to pass these parameters, but that would be a good idea. 💚 

I will implement this feature in the next release.

**Reply by kadlinobit** [reply]
Great thanks. I was also looking for a method to update the user data (if someone e.g. changes username, we should be able to refresh the frontend user data), but I guess there is no option to do that yet? 

I like the way this is handled by the **nuxt/auth** module. There is an endpoint config for the most important endpoints:
```js
endpoints: {
  login: { url: '/auth/login', method: 'post' },
  refresh: { url: '/auth/refresh', method: 'post' },
  user: { url: '/users/me?fields=*,profiles.*', method: 'get' },
  logout: { url: '/auth/logout', method: 'post' }
}
```

And there there is also a method that allows to set the user data in storage:
```js
this.$auth.setUser(user)
```

**Reply by Intevel** [reply]
I think integrating an option `autoFetch` would solve this problem. I will add this option and export the `setUser` method.
